SimTKcommon/BigMatrix/include/SimTKcommon/internal/BigMatrix.h

```diff
@@ -1040,7 +1040,9 @@ std::istream& readVectorFromStreamHelper
     openBracket = (char)ch;
     if      (openBracket=='(') {in.get(); closeBracket = ')';}
     else if (openBracket=='[') {in.get(); closeBracket = ']';}
-    else lookForCloser = false;
+    else {lookForCloser = false;
+          closeBracket = 'e'; // Just to get rid of [-Wmaybe-uninitialized] warning.
+    }
```

Comment:
The problem is the violation of RAII on line 1036. Instead, defer the declarations until line 1040 and write something like:
```c++
char openBracket{(char)ch}, closeBracket{};
```
(That syntax is "value initialization" which is zero for numeric types.)